<launch>
<param name="use_sim_time" value="true"/>
<node name="map_server" pkg="map_server" type="map_server" args="$(find autocar)/maps/world/highway.yaml">
  <param name="frame_id" value="/map" />
</node>
<node name="stageros" pkg="stage_ros" type="stageros" args="-g $(find autocar)/maps/world/test8.world" respawn="false" output="screen">
  <param name="base_watchdog_timeout" value="0.2"/>
</node>
<node name="fake_localization_0" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_0">
    <param name="odom_frame_id" value="/robot_0/odom"/>
    <param name="base_frame_id" value="/robot_0/base_link"/>
</node>
<group ns="robot_0">
    <include file="$(find autocar)/urdf/display_redCar.launch"/>
    <param name="tf_prefix" value="robot_0" />
</group>
<node name="fake_localization_1" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_1">
    <param name="odom_frame_id" value="/robot_1/odom"/>
    <param name="base_frame_id" value="/robot_1/base_link"/>
</node>
<group ns="robot_1">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_1" />
</group>
<node name="fake_localization_2" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_2">
    <param name="odom_frame_id" value="/robot_2/odom"/>
    <param name="base_frame_id" value="/robot_2/base_link"/>
</node>
<group ns="robot_2">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_2" />
</group>
<node name="fake_localization_3" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_3">
    <param name="odom_frame_id" value="/robot_3/odom"/>
    <param name="base_frame_id" value="/robot_3/base_link"/>
</node>
<group ns="robot_3">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_3" />
</group>
<node name="fake_localization_4" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_4">
    <param name="odom_frame_id" value="/robot_4/odom"/>
    <param name="base_frame_id" value="/robot_4/base_link"/>
</node>
<group ns="robot_4">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_4" />
</group>
<node name="fake_localization_5" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_5">
    <param name="odom_frame_id" value="/robot_5/odom"/>
    <param name="base_frame_id" value="/robot_5/base_link"/>
</node>
<group ns="robot_5">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_5" />
</group>
<node name="fake_localization_6" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_6">
    <param name="odom_frame_id" value="/robot_6/odom"/>
    <param name="base_frame_id" value="/robot_6/base_link"/>
</node>
<group ns="robot_6">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_6" />
</group>
<node name="fake_localization_7" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_7">
    <param name="odom_frame_id" value="/robot_7/odom"/>
    <param name="base_frame_id" value="/robot_7/base_link"/>
</node>
<group ns="robot_7">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_7" />
</group>
<node name="fake_localization_8" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_8">
    <param name="odom_frame_id" value="/robot_8/odom"/>
    <param name="base_frame_id" value="/robot_8/base_link"/>
</node>
<group ns="robot_8">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_8" />
</group>
<node name="fake_localization_9" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_9">
    <param name="odom_frame_id" value="/robot_9/odom"/>
    <param name="base_frame_id" value="/robot_9/base_link"/>
</node>
<group ns="robot_9">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_9" />
</group>
<node name="fake_localization_10" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_10">
    <param name="odom_frame_id" value="/robot_10/odom"/>
    <param name="base_frame_id" value="/robot_10/base_link"/>
</node>
<group ns="robot_10">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_10" />
</group>
<node name="fake_localization_11" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_11">
    <param name="odom_frame_id" value="/robot_11/odom"/>
    <param name="base_frame_id" value="/robot_11/base_link"/>
</node>
<group ns="robot_11">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_11" />
</group>
<node name="fake_localization_12" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_12">
    <param name="odom_frame_id" value="/robot_12/odom"/>
    <param name="base_frame_id" value="/robot_12/base_link"/>
</node>
<group ns="robot_12">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_12" />
</group>
<node name="fake_localization_13" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_13">
    <param name="odom_frame_id" value="/robot_13/odom"/>
    <param name="base_frame_id" value="/robot_13/base_link"/>
</node>
<group ns="robot_13">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_13" />
</group>
<node name="fake_localization_14" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_14">
    <param name="odom_frame_id" value="/robot_14/odom"/>
    <param name="base_frame_id" value="/robot_14/base_link"/>
</node>
<group ns="robot_14">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_14" />
</group>
<node name="fake_localization_15" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_15">
    <param name="odom_frame_id" value="/robot_15/odom"/>
    <param name="base_frame_id" value="/robot_15/base_link"/>
</node>
<group ns="robot_15">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_15" />
</group>
<node name="fake_localization_16" pkg="fake_localization" type="fake_localization" respawn="false" ns="/robot_16">
    <param name="odom_frame_id" value="/robot_16/odom"/>
    <param name="base_frame_id" value="/robot_16/base_link"/>
</node>
<group ns="robot_16">
    <include file="$(find autocar)/urdf/display_greenCar.launch"/>
    <param name="tf_prefix" value="robot_16" />
</group>
<!-- teleop control the robot -->
<node pkg="autocar" type="controller.py" name="controller_0" respawn="false" output="screen" ns="/robot_0" >
<param name="hz" type="int" value="10" />
<param name="max_speed" type="double" value="5" />
<param name="min_speed" type="double" value="-5" />
</node>
<node pkg="autocar" type="teleop.py" name="teleop_0" respawn="false" output="screen" ns="/robot_0" >
<param name="hz" type="int" value="20" />
<param name="acc" type="double" value="1" />
<param name="yaw" type="double" value="0.25" />
</node>
<node pkg="autocar" type="highway_controller.py" name="high_controller" respawn="false" output="screen" >
<param name="hz" type="int" value="10" />
<param name="path2config" type="str" value="$(find autocar)/scripts/highway/test8.json" />
</node>
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find autocar)/maps/rviz_settings/highway_simu.rviz"/>
</launch>